tests/testprogs/usdt_semaphore_test.c

```diff
@@ -0,0 +1,34 @@
+#define _SDT_HAS_SEMAPHORES 1
```

Comment:
I made this a separate test program, as this macro forces *all* probes to have a semaphore, and we want to be able to test both semaphore and non-semaphore cases.